messages/src/taskprov.rs

```diff
@@ -0,0 +1,794 @@
+                VdafType::PRIO3SUM.encode(bytes);
+                arg.encode(bytes);
+            }
+            VdafType::Prio3Histogram(arg) => {
+                VdafType::PRIO3HISTOGRAM.encode(bytes);
+                encode_u24_items(bytes, &(), arg);
+            }
+            VdafType::Poplar1(arg) => {
+                VdafType::POPLAR1.encode(bytes);
+                arg.encode(bytes);
+            }
+        }
+    }
+
+    fn encoded_len(&self) -> Option<usize> {
+        Some(
+            4 + match self {
+                VdafType::Prio3Count => 0,
+                VdafType::Prio3Sum(arg) => arg.encoded_len()?,
+                VdafType::Prio3Histogram(arg) => 3 + arg.len() * 0u64.encoded_len()?,
+                VdafType::Poplar1(arg) => arg.encoded_len()?,
+            },
+        )
+    }
+}
+
+impl Decode for VdafType {
+    fn decode(bytes: &mut Cursor<&[u8]>) -> Result<Self, CodecError> {
+        match u32::decode(bytes)? {
+            VdafType::PRIO3COUNT => Ok(VdafType::Prio3Count),
+            VdafType::PRIO3SUM => Ok(VdafType::Prio3Sum(u8::decode(bytes)?)),
+            VdafType::PRIO3HISTOGRAM => Ok(VdafType::Prio3Histogram(decode_u24_items(&(), bytes)?)),
+            VdafType::POPLAR1 => Ok(VdafType::Poplar1(u16::decode(bytes)?)),
+            val => Err(CodecError::Other(
+                anyhow!("unexpected VdafType value {}", val).into(),
+            )),
+        }
+    }
+}
+
+/// Parameters for Differential Privacy. This is mostly unspecified at the moment.
+/// See [draft-irtf-cfrg-vdaf/#94][1] for discussion.
+///
+/// [1]: https://github.com/cfrg/draft-irtf-cfrg-vdaf/issues/94
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub struct DpConfig {
+    dp_mechanism: DpMechanism,
+}
+
+impl DpConfig {
+    pub fn new(dp_mechanism: DpMechanism) -> Self {
+        Self { dp_mechanism }
+    }
+
+    pub fn dp_mechanism(&self) -> DpMechanism {
+        self.dp_mechanism
+    }
+}
+
+impl Encode for DpConfig {
+    fn encode(&self, bytes: &mut Vec<u8>) {
+        self.dp_mechanism.encode(bytes)
+    }
+
+    fn encoded_len(&self) -> Option<usize> {
+        self.dp_mechanism.encoded_len()
+    }
+}
+
+impl Decode for DpConfig {
+    fn decode(bytes: &mut Cursor<&[u8]>) -> Result<Self, CodecError> {
+        Ok(Self {
+            dp_mechanism: DpMechanism::decode(bytes)?,
+        })
+    }
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[repr(u8)]
+#[non_exhaustive]
+pub enum DpMechanism {
+    Reserved,
+    None,
+}
+
+impl DpMechanism {
+    const RESERVED: u8 = 0;
+    const NONE: u8 = 1;
+}
+
+impl Encode for DpMechanism {
+    fn encode(&self, bytes: &mut Vec<u8>) {
+        match self {
+            DpMechanism::Reserved => DpMechanism::RESERVED.encode(bytes),
+            DpMechanism::None => DpMechanism::NONE.encode(bytes),
+        }
+    }
+
+    fn encoded_len(&self) -> Option<usize> {
+        match self {
+            DpMechanism::Reserved | DpMechanism::None => Some(1),
```

Comment:
nit: just return `Some(1)`

Reply:
I used the redundant `match` such that the statement would give a compiler error if DpMechanism is expanded, guarding against the (admittedly unlikely) scenario of someone forgetting to update `encoded_len()`. If it's too noisy I can remove it.